server/fleet/apple_mdm.go

```diff
@@ -192,3 +197,82 @@ func (e MDMAppleCommandTimeoutError) Error() string {
 func (e MDMAppleCommandTimeoutError) StatusCode() int {
 	return http.StatusGatewayTimeout
 }
+
+// Mobileconfig is the byte slice corresponding to an XML property list (i.e. plist) representation
+// of an Apple MDM configuration profile in Fleet.
+//
+// Configuration profiles are used to configure Apple devices. See also
+// https://developer.apple.com/documentation/devicemanagement/configuring_multiple_devices_using_profiles.
+type Mobileconfig []byte
+
+// ParseConfigProfile attempts to parse the Mobileconfig byte slice as a Fleet MDMAppleConfigProfile.
+//
+// The byte slice must be XML or PKCS7 parseable. Fleet also requires that it contains both
+// a PayloadIdentifier and a PayloadDisplayName and that it has PayloadType set to "Configuration".
+func (mc *Mobileconfig) ParseConfigProfile() (*MDMAppleConfigProfile, error) {
+	mcBytes := *mc
+	if len(mcBytes) > 5 && string(mcBytes[0:5]) != "<?xml" {
```

Comment:
Nit, but you can do `!bytes.HasPrefix(mcBytes, []byte("<?xml"))`, which is a bit easier to read.